src/layout/IHyprLayout.cpp

```diff
@@ -178,6 +182,54 @@ void IHyprLayout::onWindowCreatedFloating(PHLWINDOW pWindow) {
     }
 }
 
+bool IHyprLayout::onWindowCreatedAutoGroup(PHLWINDOW pWindow) {
+    static auto AUTOGROUP = CConfigValue<Hyprlang::INT>("group:auto_group");
+    if ((*AUTOGROUP || g_pInputManager->m_bWasDraggingWindow) && g_pCompositor->m_pLastWindow.lock() && g_pCompositor->m_pLastWindow != pWindow &&
+        g_pCompositor->m_pLastWindow->m_pWorkspace == pWindow->m_pWorkspace && g_pCompositor->m_pLastWindow->m_sGroupData.pNextWindow.lock() &&
+        pWindow->canBeGroupedInto(g_pCompositor->m_pLastWindow.lock()) && !g_pXWaylandManager->shouldBeFloated(pWindow)) {
+
+        switch (pWindow->m_bIsFloating) {
+            case (false):                                           // tiling mode is enabled (allfloat=false on the workspace)
+                if (!g_pCompositor->m_pLastWindow->m_bIsFloating) { // target: focused tiled group
+                    static auto USECURRPOS = CConfigValue<Hyprlang::INT>("group:insert_after_current");
+                    (*USECURRPOS ? g_pCompositor->m_pLastWindow : g_pCompositor->m_pLastWindow->getGroupTail())->insertWindowToGroup(pWindow);
+                }
+
+                if (g_pCompositor->m_pLastWindow->m_bIsFloating) { // target: focused floating group
+                    // make the new tiling window to float before merging it into the focused floating group
+                    pWindow->m_bIsFloating = true;
+                    g_pLayoutManager->getCurrentLayout()->onWindowCreated(pWindow);
```

Comment:
isnt this potentially infinite recursion?

Reply:
from here it goes back into `case (true)` and group. I can't think of a scenario where it would get stuck into infinite recursion.

Reply:
if you want we can remove the recursion though